sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -31,9 +31,11 @@ class DDLSuite extends QueryTest with SharedSQLContext {
    */
   private def withDatabase(dbNames: String*)(f: => Unit): Unit = {
     try f finally {
+      val currentDb = sqlContext.sessionState.catalog.getCurrentDatabase
       dbNames.foreach { name =>
         sqlContext.sql(s"DROP DATABASE IF EXISTS $name CASCADE")
       }
+      sqlContext.sessionState.catalog.setCurrentDatabase(currentDb)
```

Comment:
How about doing it directly? `setCurrentDatabase("default")`

Then, you do not need to first get the value for `currentDb` since `currentDb` could be any database name here.